<a href="https://colab.research.google.com/github/Nikolai0Huarcaya/C8280/blob/main/Actividad1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **CLI USO DE COMANDOS Y PANDAS - VICTOR NIKOLAI HUARCAYA PUMACAYO**

**USUSANDO COMANDOS**

**Descargando un paquete para poder trabajar**

In [36]:
!apt-get install -y curl


Reading package lists... Done
Building dependency tree       
Reading state information... Done
curl is already the newest version (7.68.0-1ubuntu2.18).
0 upgraded, 0 newly installed, 0 to remove and 38 not upgraded.


**Llamamos al data.tsv**

In [37]:
!head data.tsv


order_id	quantity	item_name	choice_description	item_price
1	1	Chips and Fresh Tomato Salsa	NULL	$2.39 
1	1	Izze	[Clementine]	$3.39 
1	1	Nantucket Nectar	[Apple]	$3.39 
1	1	Chips and Tomatillo-Green Chili Salsa	NULL	$2.39 
2	2	Chicken Bowl	[Tomatillo-Red Chili Salsa (Hot), [Black Beans, Rice, Cheese, Sour Cream]]	$16.98 
3	1	Chicken Bowl	[Fresh Tomato Salsa (Mild), [Rice, Cheese, Sour Cream, Guacamole, Lettuce]]	$10.98 
3	1	Side of Chips	NULL	$1.69 
4	1	Steak Burrito	[Tomatillo Red Chili Salsa, [Fajita Vegetables, Black Beans, Pinto Beans, Cheese, Sour Cream, Guacamole, Lettuce]]	$11.75 
4	1	Steak Soft Tacos	[Tomatillo Green Chili Salsa, [Pinto Beans, Cheese, Sour Cream, Lettuce]]	$9.25 


**Ordenando el data.tsv**


In [ ]:
!cat data.tsv | column -t -s $'\t'
#Nos muestra todos los datos del data.tsv

**Llamando solo 2 filas de nuestro data.tsv**

In [24]:
!head -n 2 data.tsv | column -t -s $'\t'


order_id  quantity  item_name                     choice_description  item_price
1         1         Chips and Fresh Tomato Salsa  NULL                $2.39 


**Llamando las columnas**

In [23]:
!head -n 1 data.tsv


order_id	quantity	item_name	choice_description	item_price


**Pregunta 1.- Mire en la parte superior e inferior de data.tsv entregado en esta actividad. Piensa en cómo se estructuran los datos. ¿Qué crees que significa cada columna? ¿Qué crees que significa cada fila?**



*   
Respuesta 1 : Primero mencionaremos con respecto a las columnas, que son 
los datos que quedan registrados en la bas de datos una vez que son ingresados como se puede mostrar en los códigos de parte de abajo.
*   
Respuesta 2: Por otro lado las filas son las carácterísticas, como son el número total de datos, se podría ser que son la cantidad pedida por cada dato ingresado, obviamente los datos ingresados y trambién la descripción de ellos tendra un platillo o tabla que se demuestra exprese.





**2.- ¿Cuántos pedidos parece haber?**


In [3]:
!tail data.tsv


1831	1	Chips	NULL	$2.15 
1831	1	Bottled Water	NULL	$1.50 
1832	1	Chicken Soft Tacos	[Fresh Tomato Salsa, [Rice, Cheese, Sour Cream]]	$8.75 
1832	1	Chips and Guacamole	NULL	$4.45 
1833	1	Steak Burrito	[Fresh Tomato Salsa, [Rice, Black Beans, Sour Cream, Cheese, Lettuce, Guacamole]]	$11.75 
1833	1	Steak Burrito	[Fresh Tomato Salsa, [Rice, Sour Cream, Cheese, Lettuce, Guacamole]]	$11.75 
1834	1	Chicken Salad Bowl	[Fresh Tomato Salsa, [Fajita Vegetables, Pinto Beans, Guacamole, Lettuce]]	$11.25 
1834	1	Chicken Salad Bowl	[Fresh Tomato Salsa, [Fajita Vegetables, Lettuce]]	$8.75 
1834	1	Chicken Salad Bowl	[Fresh Tomato Salsa, [Fajita Vegetables, Pinto Beans, Lettuce]]	$8.75 



Con el comando anterior podemos saber que hay un total de 1833 pedidos 

**3.-¿Cuántas líneas hay en este archivo?**


In [8]:
!wc -l data.tsv


4624 data.tsv


Con el comando anterior podemos saber que en el archivo se encuentran 4622 líneas.

**4.-¿Qué burrito es más popular, steak o chicken?**

**Primero hacemos un código para steak para comparlo después con chicken**


In [9]:
#Steak
!grep -i 'steak burrito' data.tsv | wc -l


368


In [10]:
#Chicken
!grep -i 'chicken burrito' data.tsv | wc -l

553


Con los códigos que pusimos anteriormente y con la salida de cada uno, se afirma que **el burrito más popular es chicken con 553**.

**5.-¿Los burritos de pollo suelen tener frijoles negros o frijoles pintos?**

In [25]:
#burritos de pollo con frijoles negros
!grep -i 'chicken burrito' data.tsv | grep -i 'black beans' | wc -l

282


In [26]:
#burritos de pollo con frijoles pintos
!grep -i 'chicken burrito' data.tsv | grep -i 'pinto beans' | wc -l

105


Con esto podemos decir que los burritos de pollo con frijoles negros suelen tener 282 y los burritos de pollo con frijoles pintos suelen a tener 105

**6.-Usa la línea de comandos para descubrir algo interesante sobre los datos dados.**

In [41]:
!head -n 10 data.tsv


order_id	quantity	item_name	choice_description	item_price
1	1	Chips and Fresh Tomato Salsa	NULL	$2.39 
1	1	Izze	[Clementine]	$3.39 
1	1	Nantucket Nectar	[Apple]	$3.39 
1	1	Chips and Tomatillo-Green Chili Salsa	NULL	$2.39 
2	2	Chicken Bowl	[Tomatillo-Red Chili Salsa (Hot), [Black Beans, Rice, Cheese, Sour Cream]]	$16.98 
3	1	Chicken Bowl	[Fresh Tomato Salsa (Mild), [Rice, Cheese, Sour Cream, Guacamole, Lettuce]]	$10.98 
3	1	Side of Chips	NULL	$1.69 
4	1	Steak Burrito	[Tomatillo Red Chili Salsa, [Fajita Vegetables, Black Beans, Pinto Beans, Cheese, Sour Cream, Guacamole, Lettuce]]	$11.75 
4	1	Steak Soft Tacos	[Tomatillo Green Chili Salsa, [Pinto Beans, Cheese, Sour Cream, Lettuce]]	$9.25 


En este comando solo se esta llamando solo 10 filas del data.tsv

**USANDO PANDAS**

**Pregunta 1.- Mire en la parte superior e inferior de data.tsv entregado en esta actividad. Piensa en cómo se estructuran los datos. ¿Qué crees que significa cada columna? ¿Qué crees que significa cada fila?**
*   Respuesta 1 : Primero mencionaremos con respecto a las columnas, que son los datos que quedan registrados en la bas de datos una vez que son ingresados como se puede mostrar en los códigos de parte de abajo.
*   Respuesta 2: Por otro lado las filas son las carácterísticas, como son el número total de datos, se podría ser que son la cantidad pedida por cada dato ingresado, obviamente los datos ingresados y trambién la descripción de ellos tendra un platillo o tabla que se demuestra exprese.




In [ ]:
import pandas as pd
url='https://raw.githubusercontent.com/Nikolai0Huarcaya/C8280/main/Actividad1/data.tsv'

In [ ]:
Datos = pd.read_table(url)
Datos

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description item_price  
0                                                   NaN     $2.39   
1                                          [Clementine]     $3.39   
2                                               [Apple]     $3.39   
3                                                   NaN     $2.39   
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
...                                                 ...        ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25   
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75   
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75   

[4622 rows x 5 columns]

**A continuación se mostrara cuantas columnas y filas tiene este data.tsv**

In [ ]:
Datos.shape

(4622, 5)

In [ ]:
Datos.columns

Index(['order_id', 'quantity', 'item_name', 'choice_description',
       'item_price'],
      dtype='object')

In [ ]:
columnas = Datos.columns
columnas

Index(['order_id', 'quantity', 'item_name', 'choice_description',
       'item_price'],
      dtype='object')

In [ ]:
type(columnas)

pandas.core.indexes.base.Index

**Aquí con este código podemos observar el indice, el número de columnas del dataset más especificado**


In [ ]:
Datos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4622 entries, 0 to 4621
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   order_id            4622 non-null   int64 
 1   quantity            4622 non-null   int64 
 2   item_name           4622 non-null   object
 3   choice_description  3376 non-null   object
 4   item_price          4622 non-null   object
dtypes: int64(2), object(3)
memory usage: 180.7+ KB


**En esta sección cambiamos el archivo tsv a csv que seguidamente se subio al directorio, pero como se puede obsevar no hay cambios que afecten a la ejecución del código**

In [ ]:
import pandas as pd

In [ ]:
url="https://raw.githubusercontent.com/Nikolai0Huarcaya/C8280/main/Actividad1/data.csv"

In [ ]:
datos=pd.read_csv(url)
datos

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description item_price  
0                                                   NaN     $2.39   
1                                          [Clementine]     $3.39   
2                                               [Apple]     $3.39   
3                                                   NaN     $2.39   
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
...                                                 ...        ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25   
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75   
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75   

[4622 rows x 5 columns]

**2.- ¿Cuántos pedidos parece haber?**




In [ ]:
pedidos = Datos["quantity"].sum()
print("pedidos:" , pedidos)

pedidos: 4972


Cómo podemos ver hay 4972 pedidos realizados en el data.


**3.-¿Cuántas líneas hay en este archivo?**



In [ ]:
len(Datos)

4622

Con el código podemos saber que en el archivo se encuentran 4622 líneas.

**4.-¿Qué burrito es más popular, steak o chicken?**

**Primero hacemos un código para steak para comparlo después con chicken**






In [ ]:
#Steak
burrito_counts = Datos["item_name"].value_counts()
steak_burrito = burrito_counts.get("Steak Burrito", 0)
print("Steak Burrito:", steak_burrito)

Steak Burrito: 368


In [ ]:
#Chicken
burrito_counts = Datos["item_name"].value_counts()
Chicken_burrito = burrito_counts.get("Chicken Burrito", 0)
print("Chicken Burrito:", Chicken_burrito)

Chicken Burrito: 553


Con los códigos que pusimos anteriormente y con la salida de cada uno, se afirma que **el burrito más popular es chicken con 553**.

**5.-¿Los burritos de pollo suelen tener frijoles negros o frijoles pintos?**

In [ ]:
burritos_pollo = Datos.query("item_name == 'Chicken Burrito'")


In [ ]:
frijoles_negros = burritos_pollo["choice_description"].str.contains("Black Beans", case=False).sum()
frijoles_pintos = burritos_pollo["choice_description"].str.contains("Pinto Beans", case=False).sum()

In [ ]:
print("El número total de burritos de pollo con frijoles negros es: {}".format(frijoles_negros))
print("El número total de burritos de pollo con frijoles pintos es: {}".format(frijoles_pintos))

El número total de burritos de pollo con frijoles negros es: 282
El número total de burritos de pollo con frijoles pintos es: 105


Con esto podemos decir que los burritos de pollo con frijoles negros suelen tener 282 y los burritos de pollo con frijoles pintos suelen a tener 105

**6.-Usa la línea de comandos para descubrir algo interesante sobre los datos dados.**

In [ ]:
item_price_column = Datos['item_price']
item_price_column

0        $2.39 
1        $3.39 
2        $3.39 
3        $2.39 
4       $16.98 
         ...   
4617    $11.75 
4618    $11.75 
4619    $11.25 
4620     $8.75 
4621     $8.75 
Name: item_price, Length: 4622, dtype: object

En esta ocasión usamos un comando para que solo nos de la columna de item_price